src/psyclone/psyir/symbols/datasymbol.py

```diff
@@ -327,3 +342,20 @@ def copy_properties(self, symbol_in):
         super().copy_properties(symbol_in)
         self._is_constant = symbol_in.is_constant
         self._initial_value = symbol_in.initial_value
+
+    def update_symbols_from(self, table):
```

Comment:
Naming is hard  :) some more discussion before this makes it to master

I think the intent is more clear using the "replace" verb as in this docstring, update could mean many things and I think I wouldn't know what this method does without knowing the context. Could this be "replace_symbols_from"? also I am not sure about the preposition should it be "replace_symbols_with"? or "to"/"using"?

Reply:
I've gone for `replace_symbols_using`.